packages/celer-web-app/src/app/AppDocumentProvider.tsx

```diff
@@ -17,71 +18,81 @@ import {
 import { ServiceCreator } from "./services";
 
 export type AppDocumentProviderProps = {
-    serviceCreator: ServiceCreator,
-    //still need this for local document...
-    shouldSetBundle: boolean
+	serviceCreator: ServiceCreator,
+	//still need this for local document...
+	shouldSetBundle: boolean
 }
 
 const compiler = new Compiler();
 const routeEngine = new RouteEngine();
 const mapEngine = new MapEngine();
 
-export const AppDocumentProvider: React.FC<AppDocumentProviderProps> = ({serviceCreator, shouldSetBundle, children}) => {
+export const AppDocumentProvider: React.FC<AppDocumentProviderProps> = ({ serviceCreator, shouldSetBundle, children }) => {
 	const warnNegativeVar = useExpWarnNegativeVar();
 	const enableInferCoord = useExpInferCoord();
 	const enableBetterBundler = useExpBetterBundler();
 
 	const { splitSetting } = useAppSetting();
 
-	useEffect(()=>{
+	useEffect(() => {
 		routeEngine.warnNegativeNumberEnable = warnNegativeVar;
 		routeEngine.inferCoord = enableInferCoord;
 	}, [warnNegativeVar, enableInferCoord]);
-    
+
 	const params = useParams();
-	const [status, setStatus] = useState<string|null>(null);
-	const [error, setError] = useState<string|null>(null);
-	const [routeSourceBundle, setRouteSourceBundle] = useState<SourceObject|null>(null);
-	const [routeSourceBundleString, setRouteSourceBundleString] = useState<string|null>(null);
+	const [status, setStatus] = useState<string | null>(null);
+	const [error, setError] = useState<string | null>(null);
+	const [routeSourceBundle, setRouteSourceBundle] = useState<SourceObject | null>(null);
+	const [routeSourceBundleString, setRouteSourceBundleString] = useState<string | null>(null);
 
-	useEffect(()=>{
+	useEffect(() => {
 		const service = serviceCreator(params);
-		service.start((doc, error, status)=>{
-			if(doc){
-				if(doc._globalError){
+		service.start((doc, error, status) => {
+			if (doc) {
+				if (doc._globalError) {
 					setError(doc._globalError);
 					setStatus(null);
 					setRouteSourceBundle(null);
-				}else{
+				} else {
 					setError(null);
 					setStatus(null);
 					setRouteSourceBundle(doc);
+					// Store current page in localStorge if it is a GitHub route
+					if (("user" in params) && ("repo" in params)) {
+						// Define the current url from the parameters
```

Comment:
Note that this path only gets executed when the document is loaded successfully - Is that by design?

Reply:
That was by design, yes. Would you rather see a URL get added to recent pages if the page does not load correctly? It's a very quick fix if so.

Reply:
I think it makes more sense to add recent page on visit instead of on document loaded.
For example your browser tracks history even if it can't connect to that website